src/components/calcite-radio-button/calcite-radio-button.tsx

```diff
@@ -299,79 +303,63 @@ export class CalciteRadioButton {
-    // The only difference is we're using "fixed" instead of "absolute" positioning thanks to this StackOverflow:
-    // @link https://stackoverflow.com/questions/24299567/radio-button-causes-browser-to-jump-to-the-top/24323870
-    this.input.style.opacity = "0";
-    this.input.style.position = "fixed";
-    this.input.style.zIndex = "-1";
-
-    if (this.value) {
-      this.input.value = this.value;
-    }
-    this.input.required = this.required;
-
-    if (this.el.getAttribute("title")) {
-      this.input.title = this.el.getAttribute("title");
-    } else if (this.name && this.value) {
-      this.input.title = `Radio button with name of ${this.name} and value of ${this.value}`;
-    } else {
-      this.input.title = this.guid;
-    }
-
-    this.input.type = "radio";
-    this.el.insertAdjacentElement("beforeend", this.input);
-  }
-
-  private renderLabel(): void {
-    // Rendering a calcite-label outside of Shadow DOM for accessibility and form participation
+  private renderLabel(): VNode {
     if (this.el.textContent) {
-      const textNodes = Array.from(this.el.childNodes).filter(
-        (childNode) => childNode.nodeName === "#text"
+      return (
+        <calcite-label
+          dir={getElementDir(this.el)}
+          disable-spacing
+          disabled={this.disabled}
+          scale={this.scale}
+        >
+          <slot />
+        </calcite-label>
       );
-      const labelText = textNodes.reduce(
-        (labelText, textNode) => (labelText += textNode.textContent),
-        ""
-      );
-      while (this.el.firstChild) {
-        this.el.removeChild(this.el.firstChild);
-      }
-      this.label = document.createElement("calcite-label");
-      this.label.setAttribute("dir", getElementDir(this.el));
-      this.disabled && this.label.setAttribute("disabled", "");
-      this.label.setAttribute("disable-spacing", "");
-      this.label.setAttribute("scale", this.scale);
-      this.label.appendChild(document.createTextNode(labelText));
-      this.el.appendChild(this.label);
     }
+    return <slot />;
   }
 
   render(): VNode {
+    let title;
+    if (this.el.getAttribute("title")) {
+      title = this.el.getAttribute("title");
+    } else if (this.name && this.value) {
+      title = `Radio button with name of ${this.name} and value of ${this.value}`;
+    } else {
+      title = this.guid;
+    }
     return (
       <Host
         aria-checked={this.checked.toString()}
         aria-disabled={this.disabled.toString()}
         labeled={this.el.textContent ? true : false}
       >
-        <div class="radio" />
-        <slot />
+        <input
+          aria-label={this.value || this.guid}
+          checked={this.checked}
+          disabled={this.disabled}
+          hidden={this.hidden}
+          id={`${this.guid}-input`}
+          name={this.name}
+          onBlur={this.onInputBlur.bind(this)}
```

Comment:
Binding methods here will create a new function on every render. Can you bind in the function declaration instead?

```ts
private onInputBlur = (): void => { /*...*/ }
```